testing/integration/agent_long_test.go

```diff
@@ -0,0 +1,294 @@
+	// because we need to separately fetch the PIDs, wait until everything is healthy before we look for running beats
+	require.Eventually(runner.T(), func() bool {
+		allHealthy := true
+		status, err := runner.agentFixture.ExecStatus(context.Background())
+
+		require.NoError(runner.T(), err)
+		for _, comp := range status.Components {
+			runner.T().Logf("component state: %s", comp.Message)
+			if comp.State != int(cproto.State_HEALTHY) {
+				allHealthy = false
+			}
+		}
+		return allHealthy
+	}, time.Minute*3, time.Second*20)
+
+	handles := []processWatcher{}
+
+	regex := regexp.MustCompile(`[\d]+`)
+	status, err := runner.agentFixture.ExecStatus(context.Background())
+	require.NoError(runner.T(), err)
+	// track running beats
+	// the `last 30s` metrics tend to report gauges, which we can't use for calculating a derivative.
+	// so separately fetch the PIDs
+	for _, comp := range status.Components {
+		pidStr := regex.FindString(comp.Message)
+		pid, err := strconv.ParseInt(pidStr, 10, 64)
+		require.NoError(runner.T(), err)
+
+		handle, err := sysinfo.Process(int(pid))
+		require.NoError(runner.T(), err)
+		handlesReg := new(regression.Regression)
+		handlesReg.SetObserved(fmt.Sprintf("%s handle usage", comp.Name))
+		handlesReg.SetVar(0, "time")
+
+		runner.T().Logf("created handle watcher for %s (%d)", comp.Name, pid)
+		handles = append(handles, processWatcher{handle: handle, pid: int(pid), name: comp.Name, regHandles: handlesReg})
+	}
+
+	testDuration, err := time.ParseDuration(testRuntime)
+	require.NoError(runner.T(), err)
+
+	timer := time.NewTimer(testDuration)
+	defer timer.Stop()
+
+	// time to perform a health check
+	ticker := time.NewTicker(time.Second * 10)
+	defer ticker.Stop()
+
+	done := false
+	start := time.Now()
+	for !done {
+		select {
+		case <-timer.C:
+			done = true
+		case <-ticker.C:
+			err := runner.agentFixture.IsHealthy(ctx)
+			require.NoError(runner.T(), err)
+			// for each running process, collect memory and handles
+			for _, handle := range handles {
+
+				ohc, ok := handle.handle.(types.OpenHandleCounter)
+				if ok {
+					handleCount, err := ohc.OpenHandleCount()
+					require.NoError(runner.T(), err)
+					handle.regHandles.Train(regression.DataPoint(float64(handleCount), []float64{time.Since(start).Seconds()}))
+				}
+
+			}
+		}
+	}
+
+	// we're measuring the handle usage as y=mx+b
+	// if the slope is increasing above a certain rate, fail the test
+	// A number of factors can change the slope during a test; shortened runtime (lots of handles allocated in the first few seconds, producing an upward slope),
+	// filebeat trying to open a large number of log files, etc
+	handleSlopeFailure := 0.1
+
+	for _, handle := range handles {
+		err = handle.regHandles.Run()
+		require.NoError(runner.T(), err)
+
+		runner.T().Logf("=============================== %s (%d)", handle.name, handle.pid)
+		runner.T().Logf("handle formula: %v", handle.regHandles.Formula)
+		coeffs := handle.regHandles.GetCoeffs()
```

Comment:
It may be worth writing some small sanity tests that your usage of this regression package is correct.

You could put positive, zero, and negative slop data into it in a unit test as a demonstration that it doesn't break. 

Our confidence in the test completely depends on this so it is probably worth a sanity check.